src/Illuminate/Database/Query/Builder.php

```diff
@@ -221,7 +221,21 @@ public function __construct(ConnectionInterface $connection,
      */
     public function select($columns = ['*'])
     {
-        $this->columns = is_array($columns) ? $columns : func_get_args();
+        $this->columns = [];
+
+        $columns = is_array($columns) ? $columns : func_get_args();
+
+        foreach ($columns as $as => $column) {
+            if (is_string($as) && (
```

Comment:
What's a case where the key is a string but the column is *not* a subquery?

Someone using `select()` with an associative array (for whatever reason)? `$query->select(['foo' => 'bar'])`

Reply:
Good question.

Biggest reason I am doing the checks is for safety. My concern was that without the additional checks, someone may accidentally send user provided data as the value (`$column`) with a named key, and then it would hit the subquery logic. So, for now, to be extra safe, I check to make sure it's either a closure or query builder instance.

That said, in the future we could potentially use this as a way of doing aliasing for all columns, not only subqueries. However, that was outside of the scope of this PR.

Reply:
Aliasing "normal" columns this way is a great idea. I think we should consider it for this PR since we are already working in the code.